dev_docs/key_concepts/encrypted_saved_objects.mdx

```diff
@@ -0,0 +1,405 @@
+        },
+      },
+    },
+    management: {
+      importableAndExportable: false,
+    },
+    modelVersions: adHocRunParamsModelVersions,
+  });
+```
+And here is what the Model Version and schemas look like:
+
+```ts
+  const adHocRunParamsModelVersions: CustomSavedObjectsModelVersionMap = {
+    '1': {
+      changes: [],
+      schemas: {
+        forwardCompatibility: rawAdHocRunParamsSchemaV1.extends({}, { unknowns: 'ignore' }),
+        create: rawAdHocRunParamsSchemaV1,
+      },
+      isCompatibleWithPreviousVersion: () => true,
+    },
+  };
+
+  const rawAdHocRunParamsSchema = schema.object({
+    apiKeyId: schema.string(),
+    apiKeyToUse: schema.string(),
+    createdAt: schema.string(),
+    duration: schema.string(),
+    enabled: schema.boolean(),
+    end: schema.maybe(schema.string()),
+    rule: rawAdHocRunParamsRuleSchema,
+    spaceId: schema.string(),
+    start: schema.string(),
+    status: rawAdHocRunStatus,
+    schedule: schema.arrayOf(rawAdHocRunSchedule),
+  });
+
+  const rawAdHocRunParamsRuleSchema = schema.object({
+    name: schema.string(),
+    tags: schema.arrayOf(schema.string()),
+    alertTypeId: schema.string(),
+    params: schema.recordOf(schema.string(), schema.maybe(schema.any())),
+    apiKeyOwner: schema.nullable(schema.string()),
+    apiKeyCreatedByUser: schema.maybe(schema.nullable(schema.boolean())),
+    consumer: schema.string(),
+    enabled: schema.boolean(),
+    schedule: schema.object({
+      interval: schema.string(),
+    }),
+    createdBy: schema.nullable(schema.string()),
+    updatedBy: schema.nullable(schema.string()),
+    updatedAt: schema.string(),
+    createdAt: schema.string(),
+    revision: schema.number(),
+  });
+
+  const rawAdHocRunStatus = schema.oneOf([
+    schema.literal('complete'),
+    schema.literal('pending'),
+    schema.literal('running'),
+    schema.literal('error'),
+    schema.literal('timeout'),
+  ]);
+
+  const rawAdHocRunSchedule = schema.object({
+    interval: schema.string(),
+    status: rawAdHocRunStatus,
+    runAt: schema.string(),
+  });
+```
+
+### Registering the type as an Encrytped Saved Object
+
+```ts
+  encryptedSavedObjects.registerType({
+    type: AD_HOC_RUN_SAVED_OBJECT_TYPE, // 'ad_hoc_run_params' - the type name must match
+    attributesToEncrypt: new Set(['apiKeyToUse']),
+    attributesToIncludeInAAD: new Set(['rule', 'spaceId']),
+  });
+```
+
+There is only one encrypted attribute, `apiKeyToUse`, which contains the API key that needs to be protected. The attributes to include in AAD consist of the `spaceId`,
+and the `rule` definition. Since the object's namespace type is `'multiple-isolated'`, the space ID (or namespace) will not automatically be part of AAD by way of the
+Saved Object Descriptor (see <DocLink id="kibDevDocsEncryptedSavedObjectsIntro" section="aad" text="AAD"/>). In theory, the space id will never change for a
+`'multiple-isolated'`, which makes it a reasonable candidate for an AAD field.
```

Comment:
I also debated including some explanation regarding whether adding `spaceId` to AAD is offering any real protection in this case.

"...but it would also require some additional management to ensure it contributes to protecting the data. For example, if an object is moved to a different space maliciously (or accidentally) in order to try to gain access to its encrypted attributes, the `spaceId` attribute does not necessarily reflect the space the object resides in, and on its own provides no mechanism to guarantee the object has not moved. Some additional mechanism to override the attribute's value with the object's current namespace prior to attempting decryption would add protection."

But I don't know if this is just too much, or if what I am saying here is even feasible to implement.

cc @azasypkin @pgayvallet

Reply:
I am going to merge the doc as-is, and potentially address this in an update.